go-controller/pkg/network-attach-def-controller/network_manager.go

```diff
@@ -77,42 +79,82 @@ func (nm *networkManagerImpl) Start() error {
 
 func (nm *networkManagerImpl) Stop() {
 	controller.Stop(nm.controller)
-
-	for _, networkControllerState := range nm.networkControllers {
+	for _, networkControllerState := range nm.getAllNetworkStates() {
```

Comment:
hmm 

technically we don't need `getAllNetworkStates` because the controller should be fully stopped so no need for thread safety?

 Does no harm but it did confuse me enough to give it a thought

Reply:
yes, just to be safe and consistent